File: scala/bob/src/main/scala/bob.scala
object Bob {
  def isShout(phrase: String) {
    val shout = """([\d\W]*?)(?:[A-Z\s]{2,})[^a-z]*""".r
    phrase match { case shout(_*) => true}
}

  def isBlank(phrase: String) {

  }

  def isQuestion(phrase: String) {

  }

  def hey(phrase: String) {
    if (isShout(phrase)) {
      "Whoa, chill out!"
    } else {
      "Whatever."
    }
  }
}

File: java/bob/src/main/java/Bob.java
public class Bob {
    public String hey(String s) {
        if (isShouting(s)) {
            return "Whoa, chill out!";
        } else if (isQuestion(s)) {
            return "Sure.";
        } else if (isSilent(s)) {
            return "Fine. Be that way!";
        } else {
            return "Whatever.";
        }
    }

    private boolean isShouting(String s) {
        if ((s.toUpperCase().equals(s)) && (s.matches(".*(\\p{Alpha}+).*"))) {
            return true;
        } else {
            return false;
        }
    }

    private boolean isQuestion(String s) {
        return s.endsWith("?");
    }

    private boolean isSilent(String s) {
        return s.trim().isEmpty();
    }
}
